src/System/Hapistrano.hs

```diff
@@ -120,10 +120,14 @@ activateRelease
   -> Hapistrano ()
 activateRelease ts deployPath release = do
   rpath <- releasePath deployPath release Nothing
-  let tpath = tempSymlinkPath deployPath
-      cpath = currentSymlinkPath deployPath
-  exec (Ln ts rpath tpath) (Just release) -- create a symlink for the new candidate
-  exec (Mv ts tpath cpath) (Just release) -- atomically replace the symlink
+  isRpathExist <- doesDirExist rpath
+  if isRpathExist then do
+    let tpath = tempSymlinkPath deployPath
+        cpath = currentSymlinkPath deployPath
+    exec (Ln ts rpath tpath) (Just release) -- create a symlink for the new candidate
+    exec (Mv ts tpath cpath) (Just release) -- atomically replace the symlink
+  else
+    failWith 1 (Just "The path to the release you want to rollback to does not exist.") (Just release)
```

Comment:
@hughjfchen `activateRelease` is used in other places besides the `rollback` function. Let's change this description to be more general and include the path that is missing:
```suggestion
    failWith 1 (Just $ "The release path: " <> show rpath <> " cannot be activated.") (Just release)
```